### ISSUE TITLE
(MindSpore GPU CUDA 10.1) Concat throwing error 'Input size is mismatching'

### ISSUE THREAD

**Issue by ID56** [mindspore-ai/mindspore#28]
Hello. I uninstalled mindspore-cpu today and installed the GPU version, with CUDA 10.1, since it seemed that some operations weren't supported on CPU yet ( I opened and closed this issue a few days back: #24 ). After installing, I verified my installation with the small code sample as described in https://www.mindspore.cn/install/en#installation-verification.

![test1](https://user-images.githubusercontent.com/38450043/79051786-fef7bd80-7c53-11ea-8cfb-d76aad612e44.png)

The installation seems to be alright.

I then attempt to try the old Concat operation to see if it works now. I use the exact example used in the [documentation](https://www.mindspore.cn/api/zh-CN/master/api/python/mindspore/mindspore.ops.operations.html#mindspore.ops.operations.Concat) for Concat, but I run into an error: 'Input size is mismatching', even though the inputs are identical in shape.

![test_2](https://user-images.githubusercontent.com/38450043/79051979-226f3800-7c55-11ea-850d-04d6141a1a3d.png)

Here's the entire log in text:

> [ERROR] ME(4847,python):2020-04-12-00:29:55.991.900 [mindspore/ccsrc/kernel/gpu/gpu_kernel_factory.cc:48] CheckIOParam] op[Concat] Input size is mismatching!
> Traceback (most recent call last):
>   File "testconc.py", line 12, in <module>
>     output = op((data1, data2))
>   File "/home/mashrur/anaconda3/envs/mindenv/lib/python3.7/site-packages/mindspore/ops/primitive.py", line 141, in __call__
>     output = _run_op(self, self.name, args)
>   File "/home/mashrur/anaconda3/envs/mindenv/lib/python3.7/site-packages/mindspore/common/api.py", line 68, in wrapper
>     results = fn(*arg, **kwargs)
>   File "/home/mashrur/anaconda3/envs/mindenv/lib/python3.7/site-packages/mindspore/ops/primitive.py", line 332, in _run_op
>     output = real_run_op(obj, op_name, tuple(op_inputs), tuple(op_mask))
> RuntimeError: mindspore/ccsrc/kernel/gpu/gpu_kernel_factory.cc:48 CheckIOParam] op[Concat] Input size is mismatching!

Any help would be appreciated. Thanks in advance.

**Comment by leonwanghui** [comment]
@ID56 Sorry for the delays, for your question above, the main cause is that currently MindSpore has limited support for `Concat` ops in `PyNative` mode with GPU backend, and we recommend you update the original code in `Graph` way (like below):
```python
import numpy as np
import mindspore.nn as nn
import mindspore.ops.operations as P
from mindspore import context, Tensor

context.set_context(device_target="GPU")

class Concat(nn.Cell):
    def __init__(self):
        super(Concat, self).__init__()

        self.cat = P.Concat()
        self.data1 = Tensor(np.array([[0, 1], [2, 1]]).astype(np.int32))
        self.data2 = Tensor(np.array([[0, 1], [2, 1]]).astype(np.int32))

    def construct(self):
        return self.cat((self.data1, self.data2))

cat = Concat()
output = cat()
print(output)
```
```
[WARNING] ME(732,python):2020-04-15-02:24:17.991.697 [mindspore/ccsrc/pre_activate/pass/convert_const_input_to_tensor_input.cc:67] CreateTupleTensor] The value 0x564367f20850of tuple is not a scalar
[WARNING] ME(732,python):2020-04-15-02:24:17.991.726 [mindspore/ccsrc/pre_activate/pass/convert_const_input_to_tensor_input.cc:100] CreateTensorInput] Create tensor failed
[WARNING] ME(732,python):2020-04-15-02:24:17.991.987 [mindspore/ccsrc/session/kernel_graph.cc:221] SplitTupleValueNodeToNodeList] failed to remove the value_node ValueNode<ValueTuple> (Tensor 
shape:[[const vector][2, 2]]Int32val:[[0 1]
 [2 1]], Tensor 
shape:[[const vector][2, 2]]Int32val:[[0 1]
 [2 1]])
[[0 1]
 [2 1]
 [0 1]
 [2 1]]
```

## Cause elaboration
MindSpore supports both `Graph` and `PyNative` mode, for the single operator execution (in your case) the system will only support `PyNative` mode by default, so it would be suggested to wrapper the operator into a user-defined Class (inheritted from `nn.Cell`), then the system can be converted to `Graph` mode by default.

Does that address your problem?

**Comment by ID56** [comment]
Thanks a lot for the response! Running the code this way  prints some strange values the first time but the results come out okay on the second run. I think ultimately Ascend is currently the best choice of hardware for this, but I don't have it unfortunately.

In any case, I'm tentatively closing this issue, but can you explain what these three warnings are?

**Comment by leonwanghui** [comment]
@ID56 If you refer to the warning log below, I guess it's some bugs to be fixed in `0.1.0-alpha` release, considering it won't break the system, we will fix it in next release.

```
[WARNING] ME(1089,python):2020-04-16-01:03:26.338.925 [mindspore/ccsrc/pre_activate/pass/convert_const_input_to_tensor_input.cc:67] CreateTupleTensor] The value 0x557851848750of tuple is not a scalar
[WARNING] ME(1089,python):2020-04-16-01:03:26.338.951 [mindspore/ccsrc/pre_activate/pass/convert_const_input_to_tensor_input.cc:100] CreateTensorInput] Create tensor failed
[WARNING] ME(1089,python):2020-04-16-01:03:26.339.201 [mindspore/ccsrc/session/kernel_graph.cc:221] SplitTupleValueNodeToNodeList] failed to remove the value_node ValueNode<ValueTuple> (Tensor 
shape:[[const vector][2, 2]]Int32val:[[0 1]
 [2 1]], Tensor 
shape:[[const vector][2, 2]]Int32val:[[0 1]
 [2 1]])
```
